src/main/java/subway/line/service/LineService.java

```diff
@@ -131,13 +112,12 @@ public void deleteLineSection(final Long lineId, final Long stationId) {
             throw new IllegalArgumentException();
         }
 
-        List<Section> sections = line.getSections();
-        for (Section section : sections) {
-            if (section.getDownStation().getId().equals(stationId)) {
-                line.subtractDownStation(section.getUpStation(), section.getDistance());
-                break;
-            }
-        }
+        line.ensureRemovableSection(stationId);
+
+        final Section sectionToRemove = line.getSections().getSections().stream()
+                .filter(section -> section.getDownStation().getId().equals(stationId))
+                .findFirst().orElseThrow(() -> new IllegalArgumentException());
 
+        line.removeSection(sectionToRemove);
```

Comment:
(optional)
이 코드도 도메인으로 이동해 보시길 추천 드립니다

```suggestion
        Station station = stationRepository.findById(stationId);
        line.removeSection(station);
```

Reply:
```
final Section sectionToRemove = line.getSections().getSections().stream()
                .filter(section -> section.getDownStation().getId().equals(stationId))
                .findFirst().orElseThrow(() -> new IllegalArgumentException())
```

아하 위의 부분도 Line 에서 검증해도 무관한 부분이었네요.
removeSection 메서드로 합치겠습니다!